### DISCUSSION TITLE
Inconsistent behavior of reportUnnecessaryComparison

### DISCUSSION THREAD

**Discussion by Zzzen** [microsoft/pyright#4585] [General]
```
a = 1
b = 2
_1 = a == 2

c = True
d = False
_2 = c == d
_22 = c != d

def f():
  a = True
  b = True
  _3 = a == b
  _4 = a != b

```

I would expect `_1/_2/_22/_3/_4` to be prompted for errors because the result of comparisons can be statically determined. But actually only `_1/_2/_22` are prompted.

playground link: https://zzzen.github.io/pyright-ast-viewer/#code/FDCGAIF5wRmAjK4BMwD6MkUtVwDGSAKgE4CuApsACZIBioANgM5VrJKE7jUjUUAzcAIAUASgBcwcOGzhSlaeETQFVGWgDMWKNHhK0AFh0BCPcCA

ref:
> reportUnnecessaryComparison [boolean or string, optional]: Generate or suppress diagnostics for `==` or `!=` comparisons or other conditional expressions that are statically determined to always evaluate to False or True. Such comparisons are sometimes indicative of a programming error. The default value for this setting is `"none"`.

**Comment by erictraut** [comment]
Yes, this is by design. The `reportUnnecessaryComparison` check is selective in what types of errors it reports. We found that there are too many legitimate cases where always-true checks are used in typical Python code. This often happens, for example, in if/elif/else chains used to exhaustively check enum values. For this reason, we don't report errors for cases like `Literal[True] == Literal[True]`. It generates too much noise. The purpose of this check is to highlight uses that are likely to be bugs.
